File: 1hdsh.dart
import 'package:dio/dio.dart';

const String baseURL = "https://1hd.sh/";

Future<void> getSource(String id) async {
  final dio = Dio(BaseOptions(headers: {
    "Referer": baseURL,
    "User-Agent":
        "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36 Edg/126.0.0.0"
  }));
  final startTime = DateTime.now();

  final data1 = (await dio.get(baseURL + id)).data;

  final ajaxResp = (await dio
          .get(RegExp(r"pl_url.*(https.*?)'").firstMatch(data1)!.group(1)!))
      .data;
  final iframeData = (await dio
          .get(RegExp(r'data-id="(.*?)"').firstMatch(ajaxResp)!.group(1)!))
      .data;
  final finalData =
      (await dio.get(RegExp(r'src="(.*?)"').firstMatch(iframeData)!.group(1)!))
          .data;

  final evalMatch = RegExp(r'eval(\(f.*?\)\)\))').firstMatch(finalData);
  if (evalMatch == null) return;
  final evalFunction = evalMatch.group(1);
  var p = RegExp(r'\[\{(.*?)\}\]').firstMatch(evalFunction!)?.group(1);
  final acMatch = RegExp(r',([\d]+,[\d]+),').firstMatch(evalFunction);
  final k =
      evalFunction.split(acMatch!.group(0)!).last.split('.').first.split('|');
  final ac = acMatch.group(1)!.split(',');

  for (var c = int.parse(ac[1]); c-- > 0;) {
    if (k[c].isNotEmpty)
      p = p?.replaceAll(
          RegExp('\\b' + c.toRadixString(int.parse(ac[0])) + '\\b'), k[c]);
  }

  print(RegExp(r'"(.*?)"').firstMatch(p!)!.group(1));

  final executionTime = DateTime.now().difference(startTime);
  print('Execution time: ${executionTime.inMilliseconds} ms');
}

Future<Map> _DoodStream(String s) async {
  var doodybaseurl = "https://d0000d.com";
  var url = RegExp(r'<li class="doodstream" rel="\d+" data-video="([^"]+)">')
      .firstMatch(s)![1];
  try {
    var dood = await (await Dio().get(url!,
            options: Options(
              headers: {'Referer': baseURL},
            )))
        .data;
    var resp = await (await Dio().get(
            doodybaseurl +
                "/pass_md5/" +
                RegExp(r"\$.get\('\/pass_md5\/([^']+)'").firstMatch(dood)![1]!,
            options: Options(
              headers: {'Referer': doodybaseurl},
            )))
        .data;

    return {
      "src": dood.contains("src: data + makePlay()")
          ? resp +
              "urcodesucks?token=${RegExp(r'\?token=([^&]+)').firstMatch(dood)![1]}&expiry=${DateTime.now().millisecondsSinceEpoch}"
          : resp.data,
      "type": "video/mp4",
      "referer": doodybaseurl
    };
  } catch (e) {}
  return {"src": "not found"};
}

void main() {
  getSource("series/the-fable-02144/1-1/");
}

File: 1hdsh.js
//! It works but it's not realiable source.

import axios from "axios";
const baseURL = "https://1hd.sh/";
const GetSource = async (id) => {
  const { data } = await axios.get(baseURL + id);
  const ajaxResp = (await axios.get(data.match(/pl_url.*(https.*?)['"]/)[1]))
    .data;
  const iframe = (await axios.get(ajaxResp.match(/data-id=["'](.*?)["']/)[1]))
    .data;
  const resp = (await axios.get(iframe.match(/src=["'](.*?)["']/)[1])).data;
  const evalFuntion = resp.match(/eval(\(f.*?)\)\)\)/g).pop();
  var p = evalFuntion.match(/\[\{(.*?)\}\]/)[1];
  const acMatch = evalFuntion.match(/,([\d]+,[\d]+),/);
  const k = evalFuntion.split(acMatch[0]).pop().split(".")[0].split("|");
  const ac = acMatch ? acMatch[1].split(",") : [36, k.length];
  var [a, c] = ac;

  while (c--)
    if (k[c])
      p = p.replace(new RegExp("\\b" + c.toString(a) + "\\b", "g"), k[c]);
  console.log(p);
};

GetSource("movie/godzilla-x-kong-the-new-empire-67926/");
// console.log(
//   eval(
//     resp
//       .match(/eval(\(f.*?)\)\)\)/g)
//       .pop()
//       .replace("eval", "")
//   ).match(/file:"(.*?)"/)[1]
// );

// const animepahe = async () => {
//   const { data } = await axios.get("https://kwik.si/e/Kbic8NJPFiDy", {
//     headers: {
//       Referer: "https://animepahe.ru/",
//       Cookie: "__ddg1=;__ddg2_=;__ddgid_=;__ddgmark_=;SERVERID=",
//     },
//   });
//   const evalF = data.match(/eval(\(f.*?)\}\)\)/g).pop();
//   var p = evalF.match(/\'(.[^']*?)';/)[1].replace("\\", "");
//   const ac = evalF
//     .split(p)
//     .pop()
//     .match(/,([\d]+,[\d]+),/);
//   var k = evalF.split(ac[0]).pop().split(".")[0].replace(/'/g, "").split("|");
//   var [a, c] = ac[1].split(",");
//   console.log(p, a, c, k.length);
//   let e = (c) => {
//     console.log(c, a);
//     return (
//       (c < a ? "" : e(parseInt(c / a))) +
//       ((c %= a) > 35 ? String.fromCharCode(c + 29) : c.toString(36))
//     );
//   };

//   while (c--) {
//     if (k[c]) {
//       p = p.replace(new RegExp("\\b" + e(c) + "\\b", "g"), k[c]);
//       console.log(new RegExp("\\b" + e(c) + "\\b", "g"), k[c]);
//     }
//   }
//   console.log(p);
// };

// animepahe();
